### PR TITLE
Deprecate postgresql_lang

### LINKED ISSUE #519: Enable integration testing of postgresql_lang
##### SUMMARY
Tests for module postgresql_lang run only on CentOS 7.
Integration testing of the lang module should be enabled on all targets, to cover the various os/python/postgres/psycopg/ansible versions. 

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
postgresql_lang

##### ADDITIONAL INFORMATION

See https://github.com/ansible-collections/community.postgresql/blob/main/tests/integration/targets/postgresql_lang/tasks/main.yml#L16
```yaml
- name: Initial CI tests of postgresql_lang module
  when:
    - ansible_facts.distribution == 'CentOS'
    - ansible_facts.distribution_major_version is version ('7', '==')
```

### PR DESCRIPTION
**Pull request by elpavel**
##### SUMMARY
Deprecating the postgresql_lang module.

The lang module works only on old Postgresql versions.

The [documentation](https://www.postgresql.org/docs/12/sql-createlanguage.html) says
_As of PostgreSQL 9.1, most procedural languages have been made into “extensions”, and should therefore be installed with CREATE EXTENSION not CREATE LANGUAGE._

##### ISSUE TYPE
- Module Deprecation Pull Request

##### COMPONENT NAME
postgresql_lang

##### ADDITIONAL INFORMATION

Fixes #519

### CODE REVIEW

**Review by jchancojr — APPROVED** [review]

**Review by Andersson007 — CHANGES_REQUESTED** [review]
@elpavel thanks for starting the process!

I think we should follow the [official deprecation guide](https://docs.ansible.com/ansible/latest/dev_guide/module_lifecycle.html#deprecating-modules-and-plugins-in-a-collection), could you please take a look?

Apologies for not mentioning this before

File: plugins/modules/postgresql_lang.py [hunk 1]
```diff
@@ -12,9 +12,11 @@
 ---
 module: postgresql_lang
 short_description: Adds, removes or changes procedural languages with a PostgreSQL database
+deprecated:
+- removed_in: "4.0.0"
+- why: As of PostgreSQL 9.1, most procedural languages have been made into extensions.
+- alternative: Use M(postgresql_ext) instead.
```
Comment (Andersson007) [inline plugins/modules/postgresql_lang.py:18]:
```suggestion
- alternative: Use M(community.postgresql.postgresql_ext) instead.
```
Using a FQCN is a [requirement](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#linking-and-other-format-macros-within-module-documentation) with this type of macros

**Review by Andersson007 — COMMENTED** [review]
LGTM

one more small formatting things

File: plugins/modules/postgresql_lang.py [hunk 2]
```diff
@@ -280,6 +286,8 @@ def main():
         supports_check_mode=True,
     )
 
+    module.warn("The postgresql_lang module has been deprecated and will be removed in community.postgresql 4.0.0")
+
```
Comment (Andersson007) [inline plugins/modules/postgresql_lang.py:290]:
```suggestion
```
i think we can remove it from here

**Review by Andersson007 — CHANGES_REQUESTED** [review]

**Comment by Andersson007** [comment]
@elpavel hm, it's strange, i see deprecations made same way in c.general, for example.. i'll ask the community

**Comment by Andersson007** [comment]
@elpavel just remove the hyphens, i didn't notice them.. thanks @felixfontein !

**Review by Andersson007 — APPROVED** [review]
LGTM

**Review by Andersson007 — APPROVED** [review]

**Pull request merged**